src/sentry/static/sentry/app/components/commitLink.tsx

```diff
@@ -2,14 +2,26 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import styled from 'react-emotion';
 
+import {Repository} from 'app/types';
 import {t} from 'app/locale';
 import {getShortCommitHash} from 'app/utils';
 import Button from 'app/components/button';
 import ExternalLink from 'app/components/links/externalLink';
 import InlineSvg from 'app/components/inlineSvg';
 
+type CommitFormatterParameters = {
+  baseUrl: string;
+  commitId: string;
+};
+
+type CommitProvider = {
+  icon: string;
+  providerIds: string[];
+  commitUrl: (CommitFormatterParameters) => string;
+};
+
 // TODO(epurkhiser, jess): This should be moved into plugins.
-const SUPPORTED_PROVIDERS = [
+const SUPPORTED_PROVIDERS: Readonly<Array<CommitProvider>> = [
```

Comment:
Do we have preference yet on `Array<>` vs `Readonly<CommitProvider[]>

Reply:
In 95% of cases I prefer `[]`, but sometimes they are not great to read nested. I think this is a pretty easy search and replace, so maybe we can deal with it later after seeing some gnarly uses?

Reply:
I'm happy to switch to `[]`